plugins/inputs/dpdk/dpdk_connector.go

```diff
@@ -0,0 +1,145 @@
+// +build linux
+
+package dpdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+const maxInitMessageLength = 1024
+
+type initMessage struct {
+	Version      string `json:"version"`
+	Pid          int    `json:"pid"`
+	MaxOutputLen uint32 `json:"max_output_len"`
+}
+
+type dpdkConnector struct {
+	pathToSocket  string
+	maxOutputLen  uint32
+	messageShowed bool
+	accessTimeout time.Duration
+	connection    net.Conn
+}
+
+func newDpdkConnector(pathToSocket string, accessTimeout config.Duration) *dpdkConnector {
+	return &dpdkConnector{
+		pathToSocket:  pathToSocket,
+		messageShowed: false,
+		accessTimeout: time.Duration(accessTimeout),
+	}
+}
+
+// Connects to the socket
+// Since DPDK is a local unix socket, it is instantly returns error or connection, so there's no need to set timeout for it
+func (conn *dpdkConnector) connect() (string, error) {
+	connection, err := net.Dial("unixpacket", conn.pathToSocket)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to connect to the socket - %v", err)
+	}
+
+	result, err := conn.readMaxOutputLen(connection)
+	if err != nil {
+		return result, err
+	}
+
+	conn.connection = connection
```

Comment:
In case `readMaxOutputLen()` for some reason fails, it might become impossible to close the connection as the variable is dropped. Can we please _first_ set `conn.connection` and afterwards read the init message!? Maybe even close the connection in case of an error?

Reply:
Good catch, fixed.